BaseWeapon/src/main/java/com/github/sef24sp4/baseweapon/ProjectileControlSystem.java

```diff
@@ -0,0 +1,39 @@
+package com.github.sef24sp4.baseweapon;
+
+import com.github.sef24sp4.common.data.Coordinates;
+import com.github.sef24sp4.common.entities.ICollidableEntity;
+import com.github.sef24sp4.common.interfaces.IEntityManager;
+import com.github.sef24sp4.common.interfaces.IGameSettings;
+import com.github.sef24sp4.common.projectile.CommonProjectile;
+import com.github.sef24sp4.common.projectile.ProjectileSPI;
+import com.github.sef24sp4.common.services.IEntityProcessingService;
+
+public class ProjectileControlSystem implements IEntityProcessingService, ProjectileSPI {
+
+	@Override
+	public CommonProjectile createProjectile(ICollidableEntity shooter) {
+		//
+		BaseProjectile projectile = new BaseProjectile(shooter);
+		//Here the shape of the CommonProjectile-Polygon is defined.
+		projectile.setPolygonCoordinates(
+				new Coordinates(-1, 1),
+				new Coordinates(-1, -1),
+				new Coordinates(1, -1),
+				new Coordinates(1, 1)
+		);
+
+		//Here is the direction of the polygon shoot-aim defined.
+		projectile.setCoordinates(shooter.getCoordinates());
+		projectile.setRotation(shooter.getRotation());
+		return projectile;
+	}
+
+	@Override
+	public void process(IEntityManager entityManager, IGameSettings gameSettings) {
+		for (BaseProjectile projectile : entityManager.getEntitiesByClass(BaseProjectile.class)) {
+			double projectileSpeed = 5.0;
```

Comment:
Please also add a check with `gameSettings.isEntityWithinFrame()` method to verify that projectiles are still within frame. If they are not then remove them.